app/views/users/show.html.erb

```diff
@@ -4,9 +4,17 @@
       <div class="col-12">
         <%= image_tag @user.photo, alt: @user.name %>
         <h2><%= @user.name %></h2>
-        <% if user_signed_in? %>
+        <% if user_signed_in? && current_user.id == @user.id %>
           <%= link_to 'Edit Profile', edit_user_path(current_user), class:'btn btn-info' %>
         <% end %>
+
+        <% if current_user.pending_friend?(@user) %>
```

Comment:
- [ ] There should be no logic in views. -- Rails has built-in mechanism to deal with view-related logic which is View Helpers (app/helpers/*_helper.rb). So the common convention is to move any logic in the views to the helpers. Please take a look at [this](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers) for more information.